[Wang Yiwen]

yiwen101 opened the pull request:
# Doctor101

**Doctor101** frees your mind from having to remember things you need to do. It's:

- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is:

1. [Download it from here](https://github.com/yiwen101/ip).
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you 😉
   
And it is **FREE**!

## Features:

- Managing tasks
- Managing deadlines (coming soon)
- Reminders (coming soon)

If you're a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public static void main(String[] args) {
    // Your Java code here
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

yiwen101 committed:
Completed level0

yiwen101 committed:
Completed level1

yiwen101 committed:
Completed level2

yiwen101 committed:
Completed level3

yiwen101 committed:
almost Completed level4, tired of adding further command and write repeated code one by one, did my research and find builder pattern, will test it, so save my code before experimenting

yiwen101 committed:
middle way in my refectoring

yiwen101 committed:
another commit, but in issue with java package

yiwen101 committed:
Done refractor

yiwen101 committed:
completed error handling

yiwen101 committed:
completed level 6; actually what happened is that when I commit level 6 last time, I realised that I run the git add . at the text-ui-test folder when commiting level 5 last time, resulting in only the changes in that two folder being commited. so I revert back and recommit level 5 and level 6

yiwen101 committed:
do not know where to add as enum

yiwen101 committed:
organise into seperate packages.:wq

yiwen101 committed:
organise code into package

yiwen101 committed:
complete level 7

yiwen101 committed:
complete level 8;

yiwen101 committed:
complete oop, package

yiwen101 committed:
Merge branch 'branch-A-Gradle'

yiwen101 committed:
fix Save, which is level-7

yiwen101 committed:
unit testing

yiwen101 committed:
java-jar completed

yiwen101 committed:
add java doc

yiwen101 committed:
fix code standard issues

yiwen101 committed:
completed level 9, did some code cleaning

yiwen101 committed:
Merge branch 'branch-A-JavaDoc'

yiwen101 committed:
Merge branch 'branch-Level-9'

yiwen101 committed:
add javadoc for the findcommand

rionshocker commented on src/main/java/command/MarkCommand.java:
```diff
@@ -0,0 +1,28 @@
+import exception.DukeException;
+import exception.InvalidInputException;
+
+public class MarkCommand implements Command{
```
Could add a spacing between Command and {.

rionshocker commented on src/main/java/data/storage/Store.java:
```diff
@@ -0,0 +1,183 @@
+public class Store {
+    private static Store store = new Store();
+    TaskList tasks = new TaskList();
+    String fileName = "duke.txt";
```
Can consider making this a constant?

AlagappanRa replied:
Agreed with this comment

rionshocker reviewed (commented):
Overall a really well written code! Was easy to follow and understand what the code is doing. Just some minor changes to be made. Keep up the good work!

AlagappanRa commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,50 @@
+import ui.Invoker;
+
 public class Duke {
+    private Invoker invoker = new Invoker();
```
Could this field be made final, since it is not reassigned? 🤔

AlagappanRa commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,50 @@
+        System.out.println("____________________________________________________________");
+
+        Duke duke = new Duke();
+        duke.invoker.setCommand("list", new ListCommand());
```
I like the abstraction of the `Invoker` class and the `setCommand` function. It allows for easier extension in the future.

AlagappanRa commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,50 @@
+
+            if (input.equals("bye")) {
+                System.out.println("Bye. Hope to see you again soon!");
+                System.out.println("____________________________________________________________");
```
Could this be abstracted out to your `Invoker` class as a static variable?
```java
public static final HORIZ_LINE = "______________________________________________"
```

AlagappanRa commented on src/main/java/command/AddTaskCommand.java:
```diff
@@ -0,0 +1,34 @@
+
+
+public class AddTaskCommand implements Command {
+    protected Builder<Task> taskBuilder;
```
Nice use of the `protected` keyword. I learnt how to improve my code from you!

AlagappanRa reviewed (commented):
LGTM. Small nitpicks + Why are you taking this mod 😃 😆.

yiwen101 committed:
finish adding check style plugin

yiwen101 committed:
add gui

yiwen101 committed:
Update build.gradle

yiwen101 committed:
deleted text ui test

yiwen101 committed:
fix some bugs with gui

yiwen101 committed:
Improve code quality

Improve code quality

Add java doc to public classes in the command packages for readability.

Add “private final” when applicable to boost robustness.

Rename TaskList into Tasklist as it is one word.Deleted dead code in the Store.jave class and fix some indentation, adjust order of imports and break long lines in multiple files.

yiwen101 committed:
refractor the location of data file.\n The duke.txt file was at the root directory. To boost conhesiveness, and to enable future developer to find this file more easily, I refractor the storage pacakge such that it now will read and wirte the duke.txt in the data folder

yiwen101 committed:
Improve code Quality \n Decouble MainWindoe with the finishing message of Duke by adding static variable FINISH_Message in the Duke class, so that Mainwindow class ask for the message, rather than hardcode one locally

yiwen101 committed:
Finish code quality improvement \n abstract the greeting message of duke as static field as well. Fixed some indentation and import ordering issues and added some javaDoc

yiwen101 committed:
Add assertion for important assumptions.\nAdd assertion to ensure that file is created and task is successfully added in store package, and to ensure the response is not empty in the DialogBox class

yiwen101 committed:
Merge pull request #2 from yiwen101/branch-A-CodeQuality

Improve code quality

yiwen101 committed:
Merge pull request #3 from yiwen101/branch-A-Assertions

Add assertion for important assumptions

yiwen101 committed:
fix the bug that change in folder name not captured by git commit by quiting the IDE and recommit

yiwen101 committed:
Fix the git failing capturing taskList to tasklist renaming issue

yiwen101 committed:
rename updateDescription method as setDescriptionAtIndex in the Tasklist and storage class for greater clarity

yiwen101 committed:
debug whehter a task is marked or not is not saved by the store

yiwen101 committed:
add tag feature, support add, delete and search by tag

yiwen101 committed:
add exception for the case of adding existing tags or deleting non-existing tags

yiwen101 committed:
add appropriate response when the result list is empty for list, find, findtag commands

yiwen101 committed:
debug, now build java JAR has its local memory of tasks

yiwen101 committed:
finished writing the readMe for the github page

yiwen101 committed:
update readMe, add images

yiwen101 committed:
add crediting for reuse, and add some FQA

yiwen101 committed:
solve two bugs with gui. sovle the bug that welcoming message is blocked. solve the bug that at most 4 tasks can be shown at once

yiwen101 committed:
upadate readMe

yiwen101 committed:
add more test cases, fix outdated task cases

yiwen101 committed:
delete outdated test

yiwen101 committed:
debug jar cannot create data file when there is not data folder, and refractor launcher and main class to ui folder

yiwen101 committed:
check against the submission list, made some clearings

yiwen101 committed:
update readme, inlcude link to the release page
